docs/tutorials/algokit-template.md

```diff
@@ -0,0 +1,141 @@
+
+### AlgoKit Functionality with Templates
+
+AlgoKit provides the `algokit init` command to initialize a new project using a template. You can either pass the template name using the `-t` flag or select a template from a list. 
+
+## Key Concepts
+
+### .algokit.toml
+
+This file is the AlgoKit configuration file for this project which can be used to specify the minimum version of the AlgoKit. This is essential to ensure that projects created with your template are always compatible with the version of AlgoKit they are using.
+
+Example from `algokit-beaker-default-template`:
+```toml
+[algokit]
+min_version = "v1.1.0-beta.4"
+```
+This specifies that the template requires at least version `v1.1.0-beta.4` of AlgoKit.
+
+### Python Support: `pyproject.toml`
+
+Python projects in AlgoKit can leverage a wide range of tools for dependency management and project configuration. While Poetry and the `pyproject.toml` file are common choices, they are not the only options.
+If you opt to use Poetry, you'll rely on the pyproject.toml file to define the project's metadata and dependencies. This configuration file can utilize the Jinja templating syntax for customization.
+
+
+Example snippet from `algokit-beaker-default-template`:
+
+```toml
+[tool.poetry]
+name = "{{ project_name }}"
+version = "0.1.0"
+description = "Algorand smart contracts"
+authors = ["{{ author_name }} <{{ author_email }}>"]
+readme = "README.md"
+
+[tool.poetry.dependencies]
+python = "^3.10"
+beaker-pyteal = "^1.0.0"
+algokit-utils = "^1.3"
+python-dotenv = "^1.0.0"
+```
+
+This example shows how project metadata and dependencies are defined in `pyproject.toml`, using Jinja syntax to allow placeholders for project metadata.
+
+### TypeScript Support: `package.json`
+
+For TypeScript projects, the `package.json` file plays a similar role as `pyproject.toml` can do for Python projects. It specifies metadata about the project and lists the dependencies required for smart contract development.
+
+Example snippet:
+
+```json
+{
+  "name": "{{ project_name }}",
+  "version": "1.0.0",
+  "description": "{{ project_description }}",
+  "scripts": {
+    "build": "tsc"
+  },
+  "devDependencies": {
+    "typescript": "^4.2.4",
+    "tslint": "^6.1.3",
+    "tslint-config-prettier": "^1.18.0"
+  }
+}
+```
+
+This example shows how Jinja syntax is used within `package.json` to allow placeholders for project metadata and dependencies.
+
+### Bootstrap Option
+
+AlgoKit templates can include a bootstrap script. This script is automatically run after the project is initialized and can perform various setup tasks like installing dependencies or setting up databases.
+
+- `env`: Copies an `.env.template` file to `.env` in the current working directory and prompts for any unspecified values. This is a critical aspect of managing environment variables securely, ensuring that sensitive data doesn't accidentally end up in version control.
```

Comment:
My upcoming pr will introduce some extra behaviour here. Something like below should be a better match:

`env`: Copies an `.env*.template` file to `.env*` in the current working directory and prompts for any unspecified values. This is a critical aspect of managing environment variables securely, ensuring that sensitive data doesn't accidentally end up in version control. 

>Please note by default algokit will scan for network prefixed `.env` variables (for example `.env.localnet` and etc). If no such prefixed files are found it will then attempt to load `.env` files by default. The network prefixed env variables are useful for cases when you want to rely on [algokit deploy command](put link to deploy command documentation once its merged from https://github.com/algorandfoundation/algokit-cli/blob/deploy-command/docs/features/deploy.md)